sdk/core/core-http/lib/util/xml.ts

```diff
@@ -34,4 +33,167 @@ export function parseXML(str: string): Promise<any> {
       });
     }
   });
+  return result;
+}
+
+export async function convertAtomXmlToJson(body: string): Promise<any> {
+  const parser = new xml2js.Parser(_getDefaultSettingsForAtomXmlOperations());
+  const result = await new Promise((resolve, reject) => {
+    parser.parseString(removeBOM(body.toString()), function(err: any, parsedBody: any) {
+      if (err) {
+        reject(err);
+      } else {
+        resolve(parsedBody);
+      }
+    });
+  });
+  return result;
+}
+
+/**
+ * @ignore
+ * @param {object} content The content payload as it is to be serialized. It should include any root node(s).
+ */
+export function convertJsonToAtomXml(content: any): string {
+  content[Constants.XML_METADATA_MARKER] = { type: "application/xml" };
+
+  let doc = xmlbuilder.create();
+
+  doc = doc
+    .begin("entry", { version: "1.0", encoding: "utf-8", standalone: "yes" })
+    .att("xmlns", "http://www.w3.org/2005/Atom");
+
+  doc = doc.ele("updated", new Date().toISOString()).up();
+
+  doc = writeElementValue(doc, "content", content);
+  return doc.doc().toString();
+}
+
+/**
+ * @ignore
+ * Gets the default xml2js settings applicable for Atom based XML operations.
+ */
+function _getDefaultSettingsForAtomXmlOperations(): any {
+  const xml2jsSettings = xml2js.defaults["0.2"];
+  xml2jsSettings.normalize = false;
+  xml2jsSettings.trim = false;
+  xml2jsSettings.attrkey = "$";
+  xml2jsSettings.charkey = "_";
+  xml2jsSettings.explicitCharkey = false;
+  xml2jsSettings.explicitArray = false;
+  xml2jsSettings.ignoreAttrs = true;
+  return xml2jsSettings;
+}
+
+/**
+ * @ignore
+ * Gets the default settings applicable for general XML operations.
+ */
+function _getDefaultSettings(): any {
+  const xml2jsSettings = xml2js.defaults["0.2"];
+  xml2jsSettings.explicitArray = false;
+  xml2jsSettings.ignoreAttrs = false;
+  xml2jsSettings.explicitCharkey = false;
+  xml2jsSettings.explicitRoot = false;
+  xml2jsSettings.normalize = true;
+  return xml2jsSettings;
```

Comment:
The previous default settings were
{
explicitArray: false,
explicitCharkey: false,	
explicitRoot: false
}

We are now adding more, what is the reason behind this?

Also, using functions for default settings makes sense in terms of modular approach if it gets used in more than one place or if the code is complicated. But, we arent using either `_getDefaultSettings()` or `_getDefaultSettingsForAtomXmlOperations()` more than once, so what is the benefit of adding one level of redirection?

Reply:
These are the settings required for Atom XML serialization per investigation.
This is to guard against the default values getting changed on the xml2js parser.

I believe getDefaultSettings() already existed even though it was used just once. This was mostly to have separation of concerns.

Reply:
My question on extra settings was not on the settings being set in `_getDefaultSettingsForAtomXmlOperations()`.
It was for `_getDefaultSettings()`.

In this PR, `_getDefaultSettings()` has been introduced to be used inside the existing `parseXML()` method
This method previously only cared about `explicitArray`, `explicitCharkey` and `explicitRoot` settings.
With the changes in this PR, `parseXML()` now uses the `xml2js.defaults` and then overrides the values in it.
What was the reason for this change?
From what I could see the `parseXML()` function is not in the code path of the Atom XML serialization work.

I would expect `parseXML()` method to not be touched at all and the contents of `_getDefaultSettingsForAtomXmlOperations()` to be inlined in `convertAtomXmlToJson()`

Reply:
Ah, got it - the settings are specific to each usage of the parser and seems to be shared across instances by the xml2js library - hence explicitly setting the defaults qnd values required of each usage.

Reply:
Please see comment at https://github.com/Azure/azure-sdk-for-js/pull/4916#discussion_r331158525 and #5344, updated PR that addresses the concerns around this.